oga/backend/users/views.py

```diff
@@ -46,33 +72,35 @@ def questions(request):
         question.save()
         response_dict = {'id': question.id}
         return JsonResponse(response_dict, status=201)
+    else:
+        return HttpResponse(status=405)
 
+@check_request
 @csrf_exempt
 def sign_up(request):
     if request.method == 'POST':
-        try:
-            req_data = json.loads(request.body.decode())
-            username = req_data['username']
-            password = req_data['password']
-        except (KeyError, json.JSONDecodeError):
-            return HttpResponseBadRequest()
+        req_data = json.loads(request.body.decode())
+        username = req_data['username']
+        password = req_data['password']
         new_user = User.objects.create_user(username=username, password=password)
         response_dict = {'id': new_user.id}
         return JsonResponse(response_dict, status=201)
+    else:
+        return HttpResponseNotAllowed(['POST'])
 
+@check_request
 @csrf_exempt
 def sign_in(request):
     if request.method == 'POST':
-        try:
-            req_data = json.loads(request.body.decode())
-            username = req_data['username']
-            password = req_data['password']
-        except (KeyError, json.JSONDecodeError):
-            return HttpResponseBadRequest()
+        req_data = json.loads(request.body.decode())
+        username = req_data['username']
+        password = req_data['password']
         user = authenticate(username=username, password=password)
         if user is not None:
             login(request, user)
             response_dict = {'id': user.id}
-            return JsonResponse(response_dict, status=201)
+            return JsonResponse(response_dict, status=204)
```

Comment:
If we are not going to return anything upon successful login, please remove the response dict.
Otherwise, please change the status=204 to 201

Reply:
I changed it to status=201